luigi/contrib/lsf.py

```diff
@@ -0,0 +1,362 @@
+    import pickle
+
+import luigi
+import luigi.configuration
+from luigi.contrib.hadoop import create_packages_archive
+from luigi.contrib import lsf_runner
+from luigi.task_status import PENDING, FAILED, DONE, RUNNING, UNKNOWN
+
+"""
+LSF batch system Tasks.
+=======================
+
+What's LSF? see http://en.wikipedia.org/wiki/Platform_LSF
+and https://wiki.med.harvard.edu/Orchestra/IntroductionToLSF
+
+See: https://github.com/spotify/luigi/issues/1936
+
+This extension is modeled after the hadoop.py approach.
+I'll be making a few assumptions, and will try to note them.
+Going into it, the assumptions are:
+
+- You schedule your jobs on an LSF submission node.
+- the 'bjobs' command on an LSF batch submission system returns a standardized format.
+- All nodes have access to the code you're running.
+- The sysadmin won't get pissed if we run a 'bjobs' check every thirty
+  seconds or so per job (there are ways of coalescing the bjobs calls if that's not cool).
+
+Implementation notes:
+
+
+The procedure:
+
+- Pickle the class
+- Construct a bsub argument that runs a generic runner function with the path to the pickled class
+- Runner function loads the class from pickle
+- Runner function hits the work button on it
+
+"""
+
+LOGGER = logging.getLogger('luigi-interface')
+
+
+def attach(*packages):
+    """
+    Attach method
+    """
+    LOGGER.info("""
+        Attaching packages does nothing in LSF batch submission. All packages
+        are expected to exist on the compute node.
+    """)
+
+
+def track_job(job_id):
+    """
+    Tracking is done by requesting each job and then searching for whether the job
+    has one of the following states:
+    - "RUN",
+    - "PEND",
+    - "SSUSP",
+    - "EXIT"
+    based on the LSF documentation
+    """
+    cmd = "bjobs -noheader -o stat %d" % job_id
+    track_job_proc = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, shell=True)
+    status = track_job_proc.communicate()[0].strip('\n')
+    return status
+
+
+def kill_job(job_id):
+    """
+    Kill a running LSF job
+    """
+    subprocess.call(['bkill', job_id])
+
+
+class LSFJobTask(luigi.Task):
+    """
+    Takes care of uploading and executing an LSF job
+    """
+
+    n_cpu_flag = luigi.IntParameter(default=2, significant=False)
+    shared_tmp_dir = luigi.Parameter(default='/tmp', significant=False)
+    resource_flag = luigi.Parameter(default='mem=8192', significant=False)
+    memory_flag = luigi.Parameter(default='8192', significant=False)
+    queue_flag = luigi.Parameter(default='queue_name', significant=False)
+    runtime_flag = luigi.IntParameter(default=60)
+    job_name_flag = luigi.Parameter(default='')
+    poll_time = luigi.FloatParameter(
+        significant=False, default=5,
+        description="specify the wait time to poll bjobs for the job status")
+    save_job_info = luigi.BoolParameter(default=False)
+    output = luigi.Parameter(default="")
+    extra_bsub_args = luigi.Parameter("")
```

Comment:
I think you meant for `extra_bsub_args` to have a default empty string

Have you considered using `OptionalParameter`s with defaults of None?